File: renders/habitats/AirHabitat.cpp
/* NIM/Nama  : 13515AAA / AAA       */
/* File      : Habitat.cpp              */

#include "AirHabitat.h"

using namespace std;

AirHabitat::AirHabitat() : Habitat('O', WHITE)
{
}
AirHabitat::AirHabitat(const AirHabitat& A) : Habitat(A.id, A.color)
{
}
AirHabitat::~AirHabitat()
{
}
AirHabitat& AirHabitat::operator=(const AirHabitat& A)
{
}
char AirHabitat::render()
{
	return id;
}
Color AirHabitat::getColor()
{
	return color;
}
AirHabitat* AirHabitat::clone() const
{
	return new AirHabitat(*this);
}

File: AirHabitat.java
/**
 * @author NIM/Nama : 13515078 / Veren Iliana Kurniadi
 * File : AirHabitat.java
 */

import java.util.*;
import java.lang.*;

public class AirHabitat extends Habitat {
  /**
   * Constructor
   */
  public AirHabitat(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return 'A';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
